perimetre_asa/asaperimetre.py

```diff
@@ -38,19 +39,31 @@ class AsaPerimetre:
 
     def __init__(self):
         self.provider = None
+        self.menu = None
         self.dock = None
 
     def initProcessing(self):
         """Init Processing provider for QGIS >= 3.8."""
         self.provider = AsaPerimetreProvider()
         QgsApplication.processingRegistry().addProvider(self.provider)
+        self.dock = AsaPerimetreDock()
 
     def initGui(self):
         """Init the user interface."""
         self.initProcessing()
-        self.dock = AsaPerimetreDock()
```

Comment:
No, this must stay in the `initGui`, why in `initProcessing` ?
QGIS Desktop is calling the `initGui` to create the GUI
QGIS Server, QGIS Process are calling **only** `initProcessing` because we don't want to create GUI widgets on server

Reply:
@Gustry `self.dock = AsaPerimetreDock()` is in `initGui`

Reply:
Not before, as you can see, the line was in red